tokio-test/src/macros.rs

```diff
@@ -259,3 +259,78 @@ macro_rules! assert_err {
         }
     }};
 }
+
+/// Asserts that the expression completes within a given number of milliseconds.
+///
+/// This will invoke the `panic!` macro if the provided future
+/// expression fails to complete within the given number of
+/// milliseconds. A default timeout of 10ms is used if no duration is
+/// passed.
+///
+/// # Examples
+///
+/// ```
+/// use tokio_test::assert_completes;
+/// use tokio::time::delay_for;
+///
+/// // Succeeds because default time is longer than delay.
+/// assert_completes!(delay_for(Duration::from_millis(5)));
+///
+/// // Fails because timeout is shorter than delay.
+/// assert_completes!(delay_for(Duration::from_millis(5)), 1);
+/// ```
+#[macro_export]
+macro_rules! assert_completes {
+    ($e:expr) => {
+        assert_completes!($e, 10)
+    };
```

Comment:
Maybe this timeout should be larger?

Reply:
totally open to that, I picked ten totally arbitrarily. I actually use 50ms in my own tests, and I haven't seen any test failures that weren't related to a logic error, so I think I'll change it to that.